### PR TITLE
feat/support-tasks-in-lists

### PR DESCRIPTION
**Pull request by VxMxPx**
Support for auto inserting tasks in bullet lists (GH style).

### CODE REVIEW

**Comment by marijnh** [comment]
Would it be possible to integrate this in the regexp that's matched below your change, and add some conditional logic based on whether the task group in the regexp matched, to make it less verbose?

**Comment by VxMxPx** [comment]
Yes I can do that.

**Comment by masad-frost** [comment]
Oh what a coincidence, I just finished implementing this in our fork (which handles [this issue](https://github.com/codemirror/codemirror.next/issues/793) for us). I did exactly what Marijn is suggesting.


Here's the regex i landed on `^([ \t]*)([-+*])(?:([ \t]{1,4}\[[ xX]\])([ \t]+)|([ \t]+))`
```
capturing groups:
 1. indentation
 2. bullet style
 alt 1:
     3. list marker (including leading space)
     4. trailing spaces
 alt 2:
     5. trailing spaces
```
Changed `let after` to `let after = match[4] || match[5];` and passed the `type` from `match[2]` to `match[2] + (match[3] || '')`

**Comment by masad-frost** [comment]
Oh what a coincidence, I just finished implementing this in our fork (which handles [this issue](https://github.com/codemirror/codemirror.next/issues/793) for us). I did exactly what Marijn is suggesting.


Here's the regex i landed on `^([ \t]*)([-+*])([ \t]{1,4}\[[ xX]\])?([ \t]+)`

capturing groups:
 1. indentation
 2. bullet style
 3. list marker (including leading space)
 4. trailing spaces

Changed passed the `type` from `match[2]` to `match[2] + (match[3] || '')`

File: src/commands.ts [hunk 1]
```diff
@@ -49,6 +49,12 @@ function getContext(node: SyntaxNode, line: string, doc: Text) {
       if (after.length >= 4) { after = after.slice(0, after.length - 4); len -= 4 }
       pos += len
       context.push(new Context(node.parent!, start, pos, match[1], after, match[2], node))
+    } else if (node.name == "ListItem" && node.parent!.name == "BulletList" &&
+               (match = /^([ \t]*)([-+*][ \t]+\[[ xX]\])([ \t]+)/.exec(nodeStart(node, doc)))) {
+      let after = match[3], len = match[0].length
+      if (after.length > 8) { after = after.slice(0, after.length - 8); len -= 8 }
```
Comment (masad-frost) [inline src/commands.ts:55]: Why is this 8?

**Comment by marijnh** [comment]
Great. Want to submit your code as a new PR so I can merge it in?

**Comment by masad-frost** [comment]
Can do, will push it up tomorrow or after at most

**Pull request closed without merging**